### DISCUSSION TITLE
stfpm model

### DISCUSSION THREAD

**Discussion by mouxinyue1** [open-edge-platform/anomalib#319] [Q&A] [answered]
hello，I would like to ask why my model is STFPM and MAX_EPOCH  chooses 100, but the epoch will be random during training, sometimes ending in 13 epoch and sometimes 16 epoch？

**Comment by djdameln** [comment, marked as answer]
Hi, this is because of early stopping. The performance of the model on the validation set is being computed after each epoch, and once the performance stops improving, training will be stopped. This prevents overfitting and reduces the training time. Early stopping can be [configured](https://github.com/openvinotoolkit/anomalib/blob/development/anomalib/models/stfpm/config.yaml#L34) from the `config.yaml`. The parameters are:
- patience: How many epochs without an improvement before early stopping will be triggered.
- metric: Which performance metric should be monitored as stopping criterion.
- mode [min/max]: Specifies if the monitored metric should be minimized or maximized.

**Reply by chenmo2** [reply]
你好，那个验证集无法正常加载进行验证怎么解决这个问题，
Validation: 0it [00:00, ?it/s]2023-09-26 15:47:41,102 - anomalib.models.padim.lightning_model - INFO - Fitting a Gaussian to the embedding collected from the training set.

Validation:   0%|          | 0/12 [00:00<?, ?it/s]
Validation DataLoader 0:   0%|          | 0/12 [00:00<?, ?it/s]

**Comment by mouxinyue1** [comment]
Thank you very much for your help

**Reply by chenmo2** [reply]
Hello, that validation set is not loading properly for validation How to solve this problem,
Validation: 0it [00:00, ?it/s]2023-09-26 15:47:41,102 - anomalib.models.padim.lightning_model - INFO -将高斯拟合到从训练集中收集的嵌入。

验证：0%| | 0/12 [00:00<?, ?it/s]
验证数据加载器 0: 0%| | 0/12 [00:00<?, ?it/s]
